src/Sylius/Component/Promotion/Repository/PromotionCouponRepositoryInterface.php

```diff
@@ -32,4 +33,12 @@ public function createQueryBuilderByPromotionId($promotionId);
      * @return int
      */
     public function countByCodeLength($codeLength);
+
+    /**
+     * @param string $promotionCode
+     * @param string $code
+     *
+     * @return PromotionInterface
+     */
+    public function findOneByPromotionCodeAndCode($promotionCode, $code);
```

Comment:
But considering naming - the name of a repository method should be simple and understandable. For me `findOneByPromotionCodeAndCode` sounds just... unclear. :)

Furthermore, repository methods definitely should not be designed for one, specific use case as you've been suggesting.